### PR TITLE
Feat/binded

### PR DESCRIPTION
**Pull request by iamstarkov**

### CODE REVIEW

**Comment by iamstarkov** [comment]
as far as builds will green, lets discuss a name

File: src/binded.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const methodReducer = obj => (state, methodName) =>
+  Object.assign({}, state, { [methodName]: obj[methodName].bind(obj) });
```
Comment (kinday) [inline src/binded.js]: I’ve told you should check if `obj[methodName]` is a `Function`. Any own enumerable key of `obj` could be of any type and then you’ll get `ReferenceError` saying something has no method `bind`.

File: src/binded.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const methodReducer = obj => (state, methodName) =>
```
Comment (kinday) [inline src/binded.js]: IMHO `methodReducer` → `bind` and `binded.js` → `bindAll.js`

**Comment by iamstarkov** [comment]
i picked up `binded` because i'm using this util to get binded methods from object.

``` js
// Console methods binded to Console
const { log } = binded(console); binded
```

https://www.npmjs.com/package/binder is already taken
https://www.npmjs.com/package/binded is free
https://www.npmjs.com/package/bind-all have been taken as well

File: src/binded.js (hunk 1 above)
Comment (iamstarkov) [inline src/binded.js]: can you add an test case?

Comment (iamstarkov) [inline src/binded.js]: i'm not arguing, just need to prevent future pull-requests to not pass this check

**Comment by iamstarkov** [comment]
i got you about checking methods to be a function

**Comment by iamstarkov** [comment]
for some reason `tdd` mode is now hanging

**Comment by iamstarkov** [comment]
FYI,
- `Object.keys` doesnt work with promises,
- `Object.getOwnPropertyNames` doesnt work with node 0.10 & 0.12

**Comment by iamstarkov** [comment]
and there is something fishy with strict mode

**Comment by kinday** [comment]
> `Object.getOwnPropertyNames` doesnt work with node 0.10 & 0.12

Can you explain this?

**Comment by iamstarkov** [comment]
> > `Object.getOwnPropertyNames` doesnt work with node 0.10 & 0.12
> 
> Can you explain this?

I based my suspiciousness based on this table http://node.green/#Object-static-methods-accept-primitives, but i was wrong and i had a problem with exposed 

```
const strictModeIncompatibleProps = ['caller', 'callee', 'arguments'];
```

**Comment by iamstarkov** [comment]
> > `Object.getOwnPropertyNames` doesnt work with node 0.10 & 0.12
> 
> Can you explain this?

I based my suspiciousness based on this table http://node.green/#Object-static-methods-accept-primitives, but i was wrong and i had a problem with exposed 

```
const strictModeIncompatibleProps = ['caller', 'callee', 'arguments'];
```

**Comment by iamstarkov** [comment]
screen shot 2016-04-05 at 10 34 13

**Comment by iamstarkov** [comment]
everything operates normally with manually `bind`s replaced by this helper

**Comment by iamstarkov** [comment]
so would you say about name? i would suggest `binded` or `binded-methods`

File: src/binded.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  R.merge(state, { [name]: input[name].bind(input) });
+
+function binded(input) {
+  if (!input) {
+    throw new TypeError('`input` should not be empty');
+  }
+  return R.pipe(
+    Object.getOwnPropertyNames,
+    R.without(strictModeIncompatibleProps),
+    R.filter(R.pipe(
```
Comment (kinday) [inline src/binded.js:15]: `R.filter(R.propSatisfies(R.is(Function), R.__, input))`?

File: src/binded.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import R from 'ramda';
+
+const strictModeIncompatibleProps = ['caller', 'callee', 'arguments'];
+
+const bind = input => (state, name) =>
```
Comment (kinday) [inline src/binded.js:5]: `const bind = input => (state, name) => R.assoc(name, input[name].bind(input), state);`?

File: src/binded.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import R from 'ramda';
+
+const strictModeIncompatibleProps = ['caller', 'callee', 'arguments'];
```
Comment (kinday) [inline src/binded.js:3]: It would be nice to have annotation why you made it.

File: src/binded.js (hunk 3 above)
Comment (iamstarkov) [inline src/binded.js:15]: can you add this?

File: src/binded.js (hunk 5 above)
Comment (iamstarkov) [inline src/binded.js:3]: good idea

File: src/binded.js (hunk 4 above)
Comment (iamstarkov) [inline src/binded.js:5]: would you mind to add it as well?

File: src/binded.js (hunk 5 above)
Comment (iamstarkov) [inline src/binded.js:3]: https://github.com/iamstarkov/binded/blob/master/index.js#L3-L5

**Pull request merged**